--- xcb-proto-1.13/configure.ac.orig	2020-04-05 13:11:46.987914520 +0000
+++ xcb-proto-1.13/configure.ac	2020-04-05 13:12:26.071423560 +0000
@@ -14,9 +14,10 @@
    AC_MSG_WARN([xmllint not found; unable to validate against schema.])
 fi
 
-AM_PATH_PYTHON([2.5])
+#AM_PATH_PYTHON([2.5])
 
 xcbincludedir='${datadir}/xcb'
 AC_SUBST(xcbincludedir)
 
-AC_OUTPUT([Makefile src/Makefile xcbgen/Makefile xcb-proto.pc])
+#AC_OUTPUT([Makefile src/Makefile xcbgen/Makefile xcb-proto.pc])
+AC_OUTPUT([Makefile src/Makefile xcb-proto.pc])
